apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -311,26 +316,39 @@ export class ChangeAction extends BaseRushAction {
   }
 
   private _getChangedPackageNames(): string[] {
-    const changedFolders: (string | undefined)[] | undefined = VersionControl.getChangedFolders(
+    const changedFiles: string[] = VersionControl.getChangedFiles(
       this._targetBranch,
       this._noFetchParameter.value
     );
-    if (!changedFolders) {
+    if (changedFiles.length === 0) {
       return [];
     }
     const changedPackageNames: Set<string> = new Set<string>();
 
     const repoRootFolder: string | undefined = VersionControl.getRepositoryRootPath();
     const projectHostMap: Map<string, string> = this._generateHostMap();
 
+    const getChangeIgnorePatterns: (fsPath: string) => string[] = memoize((fsPath: string): string[] => {
+      if ((repoRootFolder && Path.isUnder(repoRootFolder, fsPath)) || !FileSystem.exists(fsPath)) {
+        return [];
+      }
+      const parentIgnoreStrings: string[] = getChangeIgnorePatterns(path.dirname(fsPath));
+
+      const changeIgnoreFilePath: string = path.join(fsPath, RushConstants.changeignoreFileName);
+
+      return FileSystem.exists(changeIgnoreFilePath)
```

Comment:
```suggestion
      try {
        return [...parentIgnoreStrings, FileSystem.readFile(changeIgnoreFilePath);
      } catch (error) {
        if (FileSystem.isNotExistsError(error)) {
          return parentIgnoreStrings;
        }
        throw error;
      }
```

Uses half as many file system operations, since `FileSystem.exists()` is internally just a try/catch around an `fs.stat` call.